llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -806,6 +784,17 @@ static VSETVLIInfo getInfoForVSETVLI(const MachineInstr &MI) {
   }
   NewInfo.setVTYPE(MI.getOperand(2).getImm());
 
+  // FIXME: store the def of AVL instead of the register in VSETVLIInfo so we
+  // don't need to peek through here with MRI.
```

Comment:
@BeMg This is the part that I'm hoping we can get rid of if we track the definition instead of the register in VSETVLIInfo. There's also something similar in `computeInfoForInstr` where we check the AVL def.